src/Agoda.IoC.NetCore.UnitTests/MicrosoftExtensionsDependencyInjectionAutowireInvalidTests.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Agoda.IoC.Core;
+using Agoda.IoC.ProjectUnderTest.Invalid12;
+using Agoda.IoC.ProjectUnderTest.Valid;
+using Microsoft.Extensions.DependencyInjection;
+using NUnit.Framework;
+using Shouldly;
+
+namespace Agoda.IoC.NetCore.UnitTests
+{
+    [TestFixture]
+    public class MicrosoftExtensionsDependencyInjectionAutowireInvalidTests
+    {
+        [Test]
+        public void RegisterByAttribute_WithAmbiguousRegistration_Throws()
+        {
+            var container = new ServiceCollection();
+            Assert.Throws<ContainerRegistrationContextException>(() => container.AutoWireAssembly(new[]
+            {
+                typeof(AmbiguousRegistration).Assembly
+            }, false,onException => {
```

Comment:
This is a bit confusing, `onException` here is the name of the parameter to the the action. The naming pattern`OnSomething()` is usually used for action handlers, not for its parameters (so, if you'd use a name method here, its signature would probably be `OnRegistrationException(ContainerRegistrationContextException exception)`, right?

Reply:
yes